src/magic.cpp

```diff
@@ -2936,8 +2937,10 @@ int spell_protection_from_evil(ubyte level, CHAR_DATA *ch, CHAR_DATA *victim, st
   // Used to identify PFE from godload_defender(), obj vnum 556
   if (skill == 150) { duration = 4; modifier = 60; } 
```

Comment:
This line should be inside of your `if(affected` line below.   This was pre-existing, but while you're in here....